ivy/functional/ivy/data_type.py

```diff
@@ -1992,15 +1996,14 @@ def promote_types(
     ret
         The type that both input types promote to
     """
+    query = [ivy.as_ivy_dtype(type1), ivy.as_ivy_dtype(type2)]
```

Comment:
As mentioned before, let's look into the duplicates in the promotion table in a different PR along with the additional logic for sorting